Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/AbstractPatientDiscoveryAuditTransforms.java

```diff
@@ -0,0 +1,216 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.patientdiscovery.audit.transform;
+
+import com.services.nhinc.schema.auditmessage.AuditMessageType;
+import com.services.nhinc.schema.auditmessage.ParticipantObjectIdentificationType;
+import gov.hhs.fha.nhinc.audit.transform.AuditTransforms;
+import gov.hhs.fha.nhinc.patientdiscovery.audit.PatientDiscoveryAuditTransformsConstants;
+import gov.hhs.fha.nhinc.patientdiscovery.parser.PRPAIN201305UV02Parser;
+import gov.hhs.fha.nhinc.patientdiscovery.parser.PRPAIN201306UV02Parser;
+import gov.hhs.fha.nhinc.transform.marshallers.JAXBContextHandler;
+import java.io.ByteArrayOutputStream;
+import java.util.List;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import org.apache.log4j.Logger;
+import org.hl7.v3.II;
+import org.hl7.v3.PRPAIN201305UV02;
+import org.hl7.v3.PRPAIN201306UV02;
+
+/**
+ * AbstractPatientDiscoveryAuditTransforms encapsulate the common functionality used by both
+ * PatientDiscoveryAuditTransforms and PatientDiscoveryDeferredRequestAuditTransforms
+ *
+ * @author tjafri
+ */
+public abstract class AbstractPatientDiscoveryAuditTransforms<T, K> extends AuditTransforms<T, K> {
+
+    private static final Logger LOG = Logger.getLogger(AbstractPatientDiscoveryAuditTransforms.class);
+    private static final String JAXB_HL7_CONTEXT_NAME = "org.hl7.v3";
+
+    protected AuditMessageType createPatientParticipantObjectIdentification(AuditMessageType auditMsg, String aa,
+        String patientId) {
+        LOG.info("createPatientParticipantObjectIdentification()");
+        ParticipantObjectIdentificationType participantObject = createParticipantObjectIdentification(
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_TYPE_CODE_SYSTEM,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_TYPE_CODE_ROLE,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_ID_TYPE_CODE,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_ID_TYPE_CODE_SYSTEM,
+            PatientDiscoveryAuditTransformsConstants.PARTICIPANT_PATIENT_OBJ_ID_TYPE_DISPLAY_NAME);
+
+        if (aa != null && patientId != null && !aa.isEmpty() && !patientId.isEmpty()) {
+            participantObject.setParticipantObjectID(createPatientId(aa, patientId));
+        }
+
+        auditMsg.getParticipantObjectIdentification().add(participantObject);
+
+        return auditMsg;
+    }
+
+    protected static String createPatientId(String assigningAuthId, String patientId) {
+        LOG.info("createPatientId()");
```

Comment:
Two comments / open questions for every place you have "LOG.info("methodName()"):

1) Should a "You Are Here ->" log statement be "info"?  This feels more like trace, or possibly debug at the most.
2) From a debugging standpoint, do we gain anything by knowing the name of the method we're in?  Keep in mind, log4j can be configured to display this info via PatternLayout params (e.g., changing to "LOG.info("xyz") would, in this case, show "[timestamp] AbstractPatientDiscoveryAuditTransforms createPatientId xyz")".